tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java

```diff
@@ -17,449 +17,502 @@
+
+import org.apache.kafka.coordinator.group.GroupCoordinatorConfig;
 import org.apache.kafka.test.TestUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
 import org.apache.kafka.tools.ToolsTestUtils;
-import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.api.extension.ExtendWith;
 
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.EnumSet;
+import java.util.HashMap;
 import java.util.HashSet;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
-import java.util.Properties;
 import java.util.List;
 import java.util.Set;
+import java.util.stream.Stream;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.stream.Collectors;
 
-import static org.apache.kafka.tools.ToolsTestUtils.TEST_WITH_PARAMETERIZED_QUORUM_AND_GROUP_PROTOCOL_NAMES;
+import static java.util.Collections.emptyMap;
 import static org.apache.kafka.common.utils.Utils.mkSet;
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertThrows;
 
-public class ListConsumerGroupTest extends ConsumerGroupCommandTest {
-    @ParameterizedTest(name = TEST_WITH_PARAMETERIZED_QUORUM_AND_GROUP_PROTOCOL_NAMES)
-    @MethodSource("getTestQuorumAndGroupProtocolParametersAll")
-    public void testListConsumerGroupsWithoutFilters(String quorum, String groupProtocol) throws Exception {
-        String simpleGroup = "simple-group";
-
-        createOffsetsTopic(listenerName(), new Properties());
-
-        addSimpleGroupExecutor(simpleGroup);
-        addConsumerGroupExecutor(1);
-        addConsumerGroupExecutor(1, PROTOCOL_GROUP, groupProtocol);
+@Tag("integration")
+@ExtendWith(ClusterTestExtensions.class)
+public class ListConsumerGroupTest {
+    private final static String TOPIC = "foo";
+    private final static String SIMPLE_GROUP = "test.simple.group";
+    private final static String DEFAULT_GROUP = "test.default.group";
+    private final static String PROTOCOL_GROUP = "test.protocol.group";
+    private final ClusterInstance clusterInstance;
+
+    ListConsumerGroupTest(ClusterInstance clusterInstance) {
+        this.clusterInstance = clusterInstance;
+    }
 
-        String[] cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list"};
-        ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs);
+    private static List<ClusterConfig> defaultGenerator() {
+        return ConsumerGroupCommandTestUtils.generator();
+    }
 
-        Set<String> expectedGroups = set(Arrays.asList(GROUP, simpleGroup, PROTOCOL_GROUP));
-        final AtomicReference<Set> foundGroups = new AtomicReference<>();
+    private static List<ClusterConfig> consumerProtocolOnlyGenerator() {
```

Comment:
How about moving this helper to `ConsumerGroupCommandTestUtils` and reuse it to rewrite `ConsumerGroupCommandTestUtils.generator`?

Reply:
Hi @chia7712, thanks for the suggestion. I found that I didn't follow old cases to test `group. coordinator. new. enable=true` with classic and consumer group protocols. I have updated the PR to follow it.

BTW, do we want to follow this comment to reduce test cases like old framework? Thanks.

https://github.com/apache/kafka/pull/15908#issuecomment-2131274897

Reply:
> BTW, do we want to follow this comment to reduce test cases like old framework? Thanks.

That is a good suggestion. Could you please file a ticket for it? We should have a separate PR for it.